src/data/bucket/symbol_bucket.js

```diff
@@ -175,7 +192,40 @@ function addCollisionBoxVertex(layoutVertexArray, point, anchor, extrude, maxZoo
  * @private
  */
 class SymbolBucket {
-    constructor(options) {
+    static programInterfaces: {
+        ['glyph' | 'icon' | 'collisionBox']: ProgramInterface
+    };
```

Comment:
I think we want `{ glyph: ProgramInterface, icon: ProgramInterface, collisionBox: ProgramInterface }` here, so that all three are required.